projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -1606,10 +1606,19 @@
 				var grandparent = parent.parentElement;
 
 				// If Gallery is made up of individual Image blocks check for custom link before
-				// loading carousel.
-				if ( grandparent && grandparent.classList.contains( 'wp-block-image' ) ) {
+				// loading carousel. The custom link may be the parent or could be a descendant
+				// of the parent if the image has rounded corners.
+				if (
+					( grandparent && grandparent.classList.contains( 'wp-block-image' ) ) ||
+					( parent && parent.classList.contains( 'wp-block-image' ) && parent.querySelector( 'a' ) )
+				) {
 					var parentHref = parent.getAttribute( 'href' );
 
+					if ( ! parentHref ) {
+						// The link must be a descendant instead.
+						parentHref = parent.querySelector( 'a' ).getAttribute( 'href' );
+					}
+
```

Comment:
While this is working, the conditional logic is not that easy to follow. As a non-blocking suggestion, it may be worth handling each branch separately, as per this suggested code.
```suggestion
				// loading carousel. The custom link may be the parent or could be a descendant
				// of the parent if the image has rounded corners.
				var parentHref = null;
				if ( grandparent && grandparent.classList.contains( 'wp-block-image' ) {
					parentHref = parent.getAttribute( 'href' );
				} else if ( parent && parent.classList.contains( 'wp-block-image' ) && parent.querySelector( 'a' ) ) {
					parentHref = parent.querySelector( 'a' ).getAttribute( 'href' );
				}
				
				if (
					parentHref &&
					parentHref.split( '?' )[ 0 ] !==
						target.getAttribute( 'data-orig-file' ).split( '?' )[ 0 ] &&
					parentHref !== target.getAttribute( 'data-permalink' )
				) {
					return;
				}
```

Reply:
Ok, let's go with this.